ansible-tests/validations/library/pacemaker.py

```diff
@@ -0,0 +1,42 @@
+import itertools
+from subprocess import Popen, PIPE
+from xml.etree import ElementTree
+
+from ansible.module_utils.basic import *
+
+
+def parse_pcs_status(pcs_status_xml):
+    root = ElementTree.fromstring(pcs_status_xml)
+    result = {
+        'failures': [dict(failure_element.items()) for failure_element
+                     in root.findall('failures/failure')],
+    }
+    return result
+
+
+def format_failure(failure):
```

Comment:
Exactly, I was suggesting you keep the keys in `parse_pcs_status`:

    result = {
        'failures': root.findall('failures/failure'),
    }

And use the key names to build your string in `format_failure`, something like:

    return "Task {task} {op_key}...".format(
        task=failure['task'],
        op_key=failure['op_key']
    )